packages/lib-file-snapshots/src/utils/test.ts

```diff
@@ -0,0 +1,71 @@
+import { rmSync } from "node:fs";
+import path from "node:path";
+import { type TestContext, expect } from "vitest";
+import type {
+  SnapshotSerializer,
+  SnapshotSerializerResult,
+} from "../types/serializer";
+import { normalizeFileName } from "./file";
+
+export const TMP_DIR = "./.tmp";
+
+export const SNAPSHOTS_DIR = "__snapshots__";
+
+export type SerializerTestFn = (context: TestContext) => Promise<void>;
+
+export function testSerializer(
+  serializer: SnapshotSerializer,
+  value: unknown,
+): SerializerTestFn {
+  return async (context): Promise<void> => {
+    const { testFileName, testName } = resolveTestContext(context);
+    const { serializedValue, fileExtension } = serializer.serialize(value);
+    const normalizedTestName = normalizeFileName(testName);
+
+    expect(serializer.canSerialize(value)).toBe(true);
+    await expect(serializedValue).toMatchFileSnapshot(
+      path.join(
+        ".",
+        SNAPSHOTS_DIR,
+        testFileName,
+        `${normalizedTestName}.${fileExtension}`,
+      ),
+    );
+  };
+}
+
+export function testSerializerThrows(
+  serializer: SnapshotSerializer,
+  value: unknown,
+): () => void {
+  return (): void => {
+    expect(serializer.canSerialize(value)).toBe(false);
+    expect(() => serializer.serialize(value)).toThrowError();
+  };
+}
+
+export class NoopSerializer implements SnapshotSerializer {
```

Comment:
Minor / naming: Just a thought. Feel free to ignore this. `NoopSerializer` does not 100% fit in my opinion. Noop sounds like the `serialize` function would do nothing, which is not allowed by design. Instead, this serializer always throws.

Reply:
You a right, the name was not the best choice. I renamed the test serializer to `FailingSerializer`.